sherpa/astro/data.py

```diff
@@ -5379,6 +5433,84 @@ def filter_region(self, data):
+    x0hi : array-like
+        Upper bound of the bins in the first dimension of the independent coordinate
+    x1hi : array-like
+        Upper bound of the bins in the second dimension of the independent coordinate
+    y : array-like
+        The values of the dependent observable.
+    shape : tuple
+        Shape of the data grid (optional). This is
+        used return the data as an image e.g. for display,
+        but it not needed for fitting and modelling.
+    staterror : array-like
+        the statistical error associated with the data
+    syserror : array-like
+        the systematic error associated with the data
+    sky : `sherpa.astro.io.wcs.WCS`
+        The optional WCS object that converts to the physical coordinate system.
+    eqpos : `sherpa.astro.io.wcs.WCS`
+        The optional WCS object that converts to the world coordinate system.
+    coord : string
+        The coordinate system of the data. This can be one of 'logical',
+        'physical', or 'world'.
+        'image' is an alias for 'logical' and 'wcs' is an alias for 'world'.
+    header : dict
+        The FITS header associated with the data to store meta data.
+
+    Example
+    -------
+    In this example, we first generate a 1000 (x,y) points from a 2D Gaussian.
+    This could be, e.g., photons observed from a star. In x direction, the center
+    of the Gaussian is at 1.2 and in y direction at 0.0. We then use
+    `~numpy.histogram2d` to bin the data into a 2D histogram. Note that numpy's
+    convention for the ordering of the axis is not the same as Sherpa's.
+    We want to use the bin edges when we create the `DataIMGInt` object, but numpy
+    gives us just one set of edges. We therefore first pick either the upper or lower
+    edges (``x0edges[:-1]`` or ``x0edges[1:]``) and then repeat them to match
+    the number of bins in the flattened data array::
+
+        >>> import numpy as np
+        >>> x = np.random.normal(size=1000, loc=1.2)
+        >>> y = np.random.normal(size=1000)
+        >>> xrange = np.arange(-2, 4.1, 0.5)
+        >>> yrange = np.arange(-2, 2.1, 0.5)
+        >>> hist, x0edges, x1edges = np.histogram2d(y, x, bins=(yrange, xrange))
+        >>> x0lo, x1lo = np.meshgrid(x0edges[:-1], x1edges[:-1])
+        >>> x0hi, x1hi = np.meshgrid(x0edges[1:], x1edges[1:])
+        >>> image = DataIMGInt("binned_image",
+        ...                    x0lo=x0lo.flatten(), x1lo=x1lo.flatten(),
+        ...                    x0hi=x0hi.flatten(), x1hi=x1hi.flatten(),
+        ...                    y=hist.flatten(), shape=hist.shape)
+
+    Note that in this example, we end up with a "logical" coordinate system
+    in ``image`` and no WCS system to convert it to anything else. On the other hand,
+    in FITS standard terminology, the "logical" coordinate system is the
+    "image", counting pixels starting at 1, while here the ``x0lo``` and ``x1lo``
+    actually start at -2. This behavior works for now, but might be revisited.
```

Comment:
Do we want to add a comment like this to the `DataMG` case above?

Reply:
done.